### PR TITLE
Remove Flex/Box prop shorthand in metabase/components

### PR DESCRIPTION
**Pull request by ariya**
This is the prerequisite before we can upgrade styled-components (see #17056).

Old code pattern (note the use of `w` as a shorthand for `width`)
```jsx
<Flex w="42px" style={{ borderRadius: 6, height: 42 }}>
```

which needs to be converted to the new code pattern:
```jsx
<Flex style={{ borderRadius: 6, width: 42, height: 42 }}>
```

*except* when the prop is array, denoting the [responsive styles](https://styled-system.com/responsive-styles/):
```jsx
const ITEM_WIDTHS = [1, 1 / 2, 1 / 3];
<Flex width={ITEM_WIDTHS} style={{ borderRadius: 6, height: 42 }}>
```

*How to verify*

1. Login to Metabase
2. Go to `/`
3. Go to `/collection/root`

With 2 and 3 above, the UI should look exactly like how it was before (in terms of spacing and padding). The modified components are being used in those two routes.

Percy-based visual tests: see #17358.

### CODE REVIEW

**Comment by ariya** [comment]
@metabase-bot run visual tests

**Comment by ariya** [comment]
@metabase-bot run visual tests

**Comment by ariya** [comment]
@metabase-bot run visual tests

**Review by alxnddr — COMMENTED** [review]
Looks good! 

Do we also need to change this occurence?
https://github.com/metabase/metabase/blob/remove-flexbox-shorthands/frontend/src/metabase/pulse/components/PulseEditCollection.jsx#L15

**Comment by ariya** [comment]
> Do we also need to change this occurence?
> https://github.com/metabase/metabase/blob/remove-flexbox-shorthands/frontend/src/metabase/pulse/components/PulseEditCollection.jsx#L15

I'm undecided about that, particularly since there is no other `style` hence it feels weird to enclose a single `width` within `style`. What do you think?

**Comment by gusaiani** [comment]
@ariya @alxnddr 

> I'm undecided about that, particularly since there is no other style hence it feels weird to enclose a single width within style. What do you think?

**tl;dr** I'd say let's convert this one to style.

### Usage of Box

In recent refactors, I've observed our code often uses `<Box>` for the wrong reason.

Afaik, `<Box>` is a div with style "box-sizing: border-box;".
Often the code uses `<Box>` just as some generic "box", meaning a "<div>", where box-sizing is irrelevant.

It's tricky though because in a few situations there are `ReactDOM.findDOMNode` operations that take box into consideration.

So we can keep the very minimal changes and worry about this later.

At the same time, we could consider: testing each Box (make it a div) to see if it breaks from the change
If it doesn't break, we could remove this Box
    
### Whether it breaks or not, we could consider just converting these to styled components right away.
This would in time makes us independent from the grid-styled lib.

I am aware of the default preference for even smaller transitions.
But I'd suggest that converting these into styled components is so trivial, it could be advantageous to just do that right away.

One example of this change, keeping the box-sizing coming from Box: https://github.com/metabase/metabase/pull/17277/files#diff-b00d5147f4f9826ae095679da3d827f56189a9c6dcd562f3f33606379060d4edR31

**Comment by ariya** [comment]
> **tl;dr** I'd say let's convert this one to style.

After looking at the remaining changes, see PR #17402, I've decided to go with the smallest possible change, mainly to decouple two risky moves: (1) migrating the shorthands (2) changing to consistent styles. If there is a regression, I want to be able to identify which one since each requires a different solution (and note that the former is tied to the planned upgraded of styled-component in #17056, while the latter isn't).

**Comment by gusaiani** [comment]
@ariya Considering what you've done in the more recent PRs, can this be merged?

Looks approvable to me…we can always do more in new PRs.

**Comment by ariya** [comment]
> @ariya Considering what you've done in more recent PRs, can this PR be merged?

That's the goal, though I still need someone's review approval :-)

**Comment by gusaiani** [comment]
> That's the goal, though I still need someone's review approval :-)

You got it 👍

**Review by kulyk — APPROVED** [review]
